### PR TITLE
randomized virus symptom stats

### PR DESCRIPTION
**Pull request by Bilbo367**
## About The Pull Request

Every round symptom stats are randomized.

## Why It's Good For The Game

This should make virology a bit better by creating more variety, and rewarding someone to get all the symptoms. Because you will be able to better leverage stats since they can be random and create better viruses.

## Changelog

:cl:
balance: Randomizes symptom stats every round.
/:cl:

### CODE REVIEW

**Comment by the-orange-cow** [comment]
People joke about viro being spreadsheet sim, but with this you would pretty much have to have an excel sheet ready to see what's even viable each shift.

**Review by MrMelbert — CHANGES_REQUESTED** [review]
(No opinion of the PR itself) You probably need to disable random stat spread in unit testing (#82518)

**Comment by LemonInTheDark** [comment]
We should remove virology

**Comment by ATH1909** [comment]
> People joke about viro being spreadsheet sim, but with this you would pretty much have to have an excel sheet ready to see what's even viable each shift.

this

randomized virus stats don't make you NOT follow a line you've planned out ahead of time, they just make that line more complex to construct

**Comment by Dawnseer** [comment]
Ah yes, I can't wait for the really vicious symptoms to boost the stats of the virus.

**Comment by Higgin** [comment]
mixed feelings.

The way viruses currently work, I can give you a top-end death virus with some shit you find in maintenance and without ever looking at a PanDEMIC.

There are about two brackets this really applies to: level 1/2 symptoms have good stats and bad effects. Level 7 symptoms have almost uniformly bad stats and terrible effects (3/5 of the symptoms are the sort of 'holy trinity' of death viruses - ARDS, AN, AP.) Level 8 are uniformly good (RCOMA/PFIX.)

Randomizing stats would make these less predictable and generally be likely to make all the same old combos less potent. In the cases where it wouldn't, it's currently entirely predictable that you get the symptoms you need to hit the thresholds to make something really awful to deal with if you know what to go after. 

Adding a bit less predictability and reliability to viruses was part of my goal in #79854 - this would be in keeping with those goals, even if you might rarely get worse or more potent combos out of it.

Honestly, it'd be worth giving this a shot.

For a bit of a picture of the current spread
Symptom_Distribution_by_Level_and_Agent_1
Virus_Statistic_Distribution_at_Level
:

I suspect this PR would make combinations using a wider variety of symptoms in the middle more viable, and viruses using the obnoxious level 1/2 and 7 symptoms less so in the preponderance. Viro is already so much RNG - it's no loss to have the numbers change round-to-round, and it doesn't need to be (nor should it if you're dealing with anything that can be spread on its own) reliable.I'd like to see this tried.

**Comment by Higgin** [comment]
> People joke about viro being spreadsheet sim, but with this you would pretty much have to have an excel sheet ready to see what's even viable each shift.

this is not a bad thing considering what's "viable" now is incredibly predictable and boring (good or bad.) 

spend a round charting out what works today and what doesn't - go with what does - is a lot more variation on both sides of the experience than me making an optimized healing virus in under 3 minutes of good RNG.

**Comment by GuillaumePrata** [comment]
There is good kind of RNG, and bad kind of RNG.

Right now, Virology has the bad kind of RNG where you already know exactly what you want, but you need to keep dropping 1u of X or Y and praying to rngesus to give you what you want already it has been 5 minutes of repetitive tasks.

Now this RNG is good, it rewards players that look at what they have in hand and come up with a plan on the fly.

We should merge this PR and then nuke the other RNG shittery out of Viro.

**Comment by Bilbo367** [comment]
> We should merge this PR and then nuke the other RNG shittery out of Viro.

I think a good fix for this would be adding in more ways to get symptoms other then the chemicals like swabbing blood, or monkies. Then we could decrease the pool size were pulling from and control what we get better.

**Comment by moocowswag** [comment]
Test merge it

**Comment by Bibarra** [comment]
I think having more leeway to make cool viruses allows for better play for antag virologists as they are less inclined to make the same viruses. Perhaps the greater issue is that there's a lack of symptoms, both positive and negative to work with? "Good" viruses have already been metagamed to death, and dangerous viruses nearly equally so, as negative symptom stats make some funny/stealthy/annoying viruses unviable. At the very least, giving virologists more options is a good thing in my view.

**Comment by GuillaumePrata** [comment]
> then down the line someone gets some god dice rolls and the server gets wiped and people remove viro.

Atmos techs can spawn, walk a few tiles and end the round with a plasmaflood.

If people really get annoyed that a late game virus that takes effort to discover each sympton, plan what would be the most dangerous mix in the middle of the RNG and then is used by an antagonist to wipe the station, then indeed virology is too much for the playerbase to handle and we should also just throw atmos in the trash bin with it.

**Comment by jughu** [comment]
> > then down the line someone gets some god dice rolls and the server gets wiped and people remove viro.
> 
> Atmos techs can spawn, walk a few tiles and end the round with a plasmaflood.
> 
> If people really get annoyed that a late game virus that takes effort to discover each sympton, plan what would be the most dangerous mix in the middle of the RNG and then is used by an antagonist to wipe the station, then indeed virology is too much for the playerbase to handle and we should also just throw atmos in the trash bin with it.

TECHNICALLY roundstart virus aslong expirimentor exists

**Comment by newfren** [comment]
this doesn't mean that I'm going to explore all the symptoms for stats to make the perfect virus every round, it means I'm going to ignore thresholds and stick random positive symptoms in instead of neutered fever and itching, and I think that's a shame

**Comment by Jackal-boop** [comment]
Let them cook. It will force viro's to do more thinking than looking for specific names then becoming a med doctor for the rest of the shift

**Comment by Formerxboxgamer** [comment]
awful, terrible RNG. Currently death symptoms are balanced around removing stats from a virus. If a death symptoms rolls good stats then you could add two death symptoms. Idk if people who already complain about death viruses want to see even faster TTK viruses because AARDS rolled a +5 to resistance.

**Comment by Higgin** [comment]
> awful, terrible RNG. Currently death symptoms are balanced around removing stats from a virus. If a death symptoms rolls good stats then you could add two death symptoms. Idk if people who already complain about death viruses want to see even faster TTK viruses because AARDS rolled a +5 to resistance.

look at the PR. 

`/datum/symptom/proc/randomize_stats()
	if(prob(50))
		return 0
	if(prob(50))
		return prob(15) + 1
	else
		return -prob(15) - 1`

at most your variation should only be +/- 2 on any given stat - but in the balance, since they generally rely on near razor margins for some of their worst thresholds, this will make death viruses less effective more often

**Comment by Higgin** [comment]
unnamed
unnamed (1)
unnamed (2)
These are some of the fastest-killing viruses I can give you right now.

Best way I can explain this for anyone looking at this PR and imagining The Great Spess Pox v2 every round:

The chance of getting positive or negative stat changes is the same for each stat on each symptom.

This is going to give us a distribution around the current normal stats (roughly the expected median and mean.) 

The worst viruses currently rely on stacking to meet high thresholds and only when you have every other slot filled with stat-boosting level 1/2 symptoms.

Increases to the values of several of these stats will be required in order to possibly remove any one of the symptoms for something else. The increases will have to be large (and thus very rare) to fit more bad symptoms besides the ones you'd be using anyway because the other hurtful symptoms all pretty much have negative stats.

Look at the examples above - with the exception of the Autophagocytosis virus, all of them rely on pretty much everything being perfect to hit their killer thresholds, and they don't really get better from extra stats.

In a world where those stat-boosting symptoms and the killer symptoms themselves can be randomly worse, they're only going to see a net decrease in usefulness across rounds because you will have rounds where these combos do not work.

Also, @Bilbo367 the power stat isn't in use on a lot of viruses, but it should probably either be reported in-game with this change or clamped so it doesn't go below 0.5 or something. Healing symptoms use it much more often.

**Comment by EricZilla** [comment]
Making viro harder instead of just a waiting sim seems like a good change to keep people coming back. As it is now it'll probably go the way of nanotites sometime next year

**Comment by Higgin** [comment]
GLOB.symptom_randomness[type]["power"] = min(0.5, (GLOB.symptom_randomness[type]["power"]))

try something like this for the second line and test it out - I should have been clearer that I mean you want to set the floor at something like 0.5 so symptoms still work even if slower, "clamp' was not right -

it's okay to have a virus be less effective for a round, but you don't want people deceived into thinking it'll have any effect for it to have no effect at all

**Comment by Bilbo367** [comment]
> GLOB.symptom_randomness[type]["power"] = min(0.5, (GLOB.symptom_randomness[type]["power"]))
> 
> try something like this for the second line and test it out - I should have been clearer that I mean you want to set the floor at something like 0.5 so symptoms still work even if slower, "clamp' was not right -
> 
> it's okay to have a virus be less effective for a round, but you don't want people deceived into thinking it'll have any effect for it to have no effect at all

Don't you mean: GLOB.symptom_randomness[type]["power"] = max(0, (GLOB.symptom_randomness[type]["power"]))

min will only turn a 1 into 0.5. The max with 0 maintains functionality since the randomness in glob is a additive measure not a multiplier so a 0 would mean 0 change.

**Comment by Higgin** [comment]
yeah, good catch - max is the right one there. I've been mobileposting and half detached for over a week or so now - I corrected what I'd originally suggested wrong.

File: code/datums/diseases/advance/symptoms/symptoms.dm [hunk 1]
```diff
@@ -41,13 +42,33 @@
 	var/required_organ
 
 /datum/symptom/New()
+	//intialize new random symptom behavior
+	if(isnull(GLOB.symptom_randomness[type]))
+		GLOB.symptom_randomness[type] = list("stealth" = randomize_stats(), "resistance" = randomize_stats(), "stage_speed" = randomize_stats(), "transmittable" = randomize_stats(), "power" = randomize_stats() / 2)
+		if(power + GLOB.symptom_randomness[type]["power"] < 0.5)
+			GLOB.symptom_randomness[type]["power"] = 0
```
Comment (Bilbo367) [inline code/datums/diseases/advance/symptoms/symptoms.dm:49]:
```suggestion
			GLOB.symptom_randomness[type]["power"] = max(0, (GLOB.symptom_randomness[type]["power"]))
```

**Comment by Higgin** [comment]
sorry but I guess I didn't make it clear what I'm suggesting - it should not be 0 if the virus has a power stat that actually does something. You don't want the symptoms to be as good as nonexistent even if they're less effective - that's a trap that the player can't access information about without in-round testing without risking it totally whiffing.

**Comment by Bilbo367** [comment]
> sorry but I guess I didn't make it clear what I'm suggesting - it should not be 0 if the virus has a power stat that actually does something. You don't want the symptoms to be as good as nonexistent even if they're less effective - that's a trap that the player can't access information about without in-round testing without risking it totally whiffing.

Sorry i took so long. IT SHOULD BE ZERO. the randomness is a additional modifier not a setter or multiplier so 0 is perfect since it makes 0 change. 0 means 0 change as I said before.

**Comment by optimumtact** [comment]
there's obviously no appetite for this change

**Pull request closed without merging**